### ISSUE TITLE
The disconnect is not working fine as the user is not redirected to the Welcome Screen

### ISSUE THREAD

**Issue by abodea** [mozilla-lockwise/lockwise-android#1012]
## Steps to reproduce
1. Launch Lockwise.
2. Login with valid credentials.
3. Browse anything on the app(make sure it works ok).
4. Reach the account settings.
5. Tap on disconnect.
### Expected behavior
The Lockwise account is dismissed and the user is redirected to the Welcome Screen where the `Get Started` button is displayed.
### Actual behavior
Nothing happens, the account screen is not dismissed and it looks like the user is still logged in(until restarting the app). 
### Device & build information
* Device: **`Google Pixel 3a XL(Android 9)`**
* Build version: **`3.0.0-rc2 (Build 5177)`**
### Notes
**Please note that I cannot reproduce this issue on the previous RC(3.0.0 (Build 5132))**
Also, note that if restarting the app, the Welcome Screen is displayed with the `Get Started` button.
Attachments: 
[Video](https://drive.google.com/file/d/10dSOl1In8c4lSQM5WjqbZTPGYxjFWp9l/view?usp=sharing)

**Comment by abodea** [comment]
```Kotlin
2019-10-22 11:07:16.955 22715-22755/? I/FirefoxAccount: Successfully executed: disconnect
2019-10-22 11:07:16.956 22715-23108/? I/FirefoxAccount: Executing: begin oauth flow
2019-10-22 11:07:16.964 1162-3850/? D/ConnectivityService: requestNetwork for uid/pid:10515/22715 NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ]
2019-10-22 11:07:16.965 1162-1426/? D/WIFI_AWARE_FACTORY: got request NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ] with score 60
2019-10-22 11:07:16.965 1162-1373/? D/WIFI: got request NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ] with score 60
2019-10-22 11:07:16.965 1162-1373/? D/WIFI_UT: got request NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ] with score 60
2019-10-22 11:07:16.966 1162-1427/? D/Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ] with score 60
2019-10-22 11:07:16.966 1759-1759/? D/PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=391, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Unwanted:  Uid: 10515] ] with score 60
2019-10-22 11:07:16.995 22715-22715/? I/Choreographer: Skipped 84 frames!  The application may be doing too much work on its main thread.
2019-10-22 11:07:16.999 22715-23201/? D/telemetry/service: Performing upload of ping type: core
2019-10-22 11:07:17.153 620-1219/? W/SurfaceFlinger: Attempting to set client state on removed layer: mozilla.lockbox/mozilla.lockbox.view.RootActivity#1
2019-10-22 11:07:17.153 620-1219/? W/SurfaceFlinger: Attempting to destroy on removed layer: mozilla.lockbox/mozilla.lockbox.view.RootActivity#1
2019-10-22 11:07:17.167 620-1219/? W/SurfaceFlinger: Attempting to set client state on removed layer: Dim Layer for - Task=83#0
2019-10-22 11:07:17.167 620-1219/? W/SurfaceFlinger: Attempting to destroy on removed layer: Dim Layer for - Task=83#0
2019-10-22 11:07:17.279 22715-23201/? D/telemetry/client: Ping upload: 200
2019-10-22 11:07:17.280 22715-23201/? D/telemetry/service: Performing upload of ping type: mobile-event
2019-10-22 11:07:17.592 22715-23201/? D/telemetry/client: Ping upload: 200
2019-10-22 11:07:17.593 22715-23201/? D/telemetry/service: All uploads performed
2019-10-22 11:07:17.633 790-972/? I/CHRE: @ 88439.492: [AR_CHRE] still: 100
2019-10-22 11:07:19.243 22715-23108/? I/FirefoxAccount: Successfully executed: begin oauth flow
2019-10-22 11:07:19.665 1759-1759/? D/QtiCarrierConfigHelper: WARNING, no carrier configs on phone Id: 0
2019-10-22 11:07:20.416 22845-22845/? I/GH.ConnLoggerV2: Stopped session 0c2f4e12-f59a-4efb-8b59-d611588b1c7a
2019-10-22 11:07:20.417 22845-22882/? D/GH.ConnLoggerV2: Session 0c2f4e12-f59a-4efb-8b59-d611588b1c7a, event 2, SESSION_STOPPED
2019-10-22 11:07:20.842 790-972/? I/CHRE: @ 88442.703: [AR_CHRE] still: 100
2019-10-22 11:07:24.043 790-972/? I/CHRE: @ 88445.906: [AR_CHRE] still: 100
2019-10-22 11:07:26.246 2033-19732/? V/NativeCrypto: Read error: ssl=0x73740649c8: I/O error during system call, Connection reset by peer
2019-10-22 11:07:26.252 2033-19732/? V/NativeCrypto: SSL shutdown failed: ssl=0x73740649c8: I/O error during system call, Broken pipe
2019-10-22 11:07:26.253 2033-19732/? E/WakeLock: GCM_HB_ALARM release without a matched acquire!
2019-10-22 11:07:26.253 2033-19732/? W/WakeLock: GCM_HB_ALARM counter does not exist
2019-10-22 11:07:26.257 1162-1380/? D/ConnectivityService: reportNetworkConnectivity(101, false) by 10021
2019-10-22 11:07:26.258 1162-8592/? D/NetworkMonitor/NetworkAgentInfo [WIFI () - 101]: Forcing reevaluation for UID 10021
2019-10-22 11:07:26.259 2033-19732/? E/WakeLock: GCM_CONN_ALARM release without a matched acquire!
2019-10-22 11:07:26.259 2033-19732/? W/WakeLock: GCM_CONN_ALARM counter does not exist
2019-10-22 11:07:26.269 1162-23312/? D/NetworkMonitor/NetworkAgentInfo [WIFI () - 101]: PROBE_DNS www.google.com 9ms OK 216.58.214.228
2019-10-22 11:07:26.312 1162-23313/? D/NetworkMonitor/NetworkAgentInfo [WIFI () - 101]: PROBE_DNS connectivitycheck.gstatic.com 52ms OK 172.217.19.99
2019-10-22 11:07:26.350 1162-23313/? D/NetworkMonitor/NetworkAgentInfo [WIFI () - 101]: PROBE_HTTP http://connectivitycheck.gstatic.com/generate_204 time=36ms ret=204 request={User-Agent=[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.32 Safari/537.36]} headers={null=[HTTP/1.1 204 No Content], Content-Length=[0], Date=[Tue, 22 Oct 2019 08:07:26 GMT], X-Android-Received-Millis=[1571731646349], X-Android-Response-Source=[NETWORK 204], X-Android-Selected-Protocol=[http/1.1], X-Android-Sent-Millis=[1571731646330]}
2019-10-22 11:07:26.390 1162-23312/? D/NetworkMonitor/NetworkAgentInfo [WIFI () - 101]: PROBE_HTTPS https://www.google.com/generate_204 time=120ms ret=204 request={User-Agent=[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.32 Safari/537.36]} headers={null=[HTTP/1.1 204 No Content], Alt-Svc=[quic=":443"; ma=2592000; v="46,43",h3-Q048=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000], Content-Length=[0], Date=[Tue, 22 Oct 2019 08:07:26 GMT], X-Android-Received-Millis=[1571731646390], X-Android-Response-Source=[NETWORK 204], X-Android-Selected-Protocol=[http/1.1], X-Android-Sent-Millis=[1571731646328]}
2019-10-22 11:07:26.393 1162-1380/? D/ConnectivityService: NetworkAgentInfo [WIFI () - 101] validation passed
2019-10-22 11:07:26.394 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.net.conn.NETWORK_CONDITIONS_MEASURED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:07:27.239 790-972/? I/CHRE: @ 88449.094: [AR_CHRE] still: 100
2019-10-22 11:07:30.435 790-972/? I/CHRE: @ 88452.305: [AR_CHRE] still: 100
2019-10-22 11:07:30.482 790-972/? I/CHRE: @ 88452.344: [ImuCal] [NanoSensorCal:GYRO_RPS] Offset | Temperature [C]: -0.000870, 0.000484, 0.000325 | 30.85
2019-10-22 11:07:30.482 790-972/? I/CHRE: @ 88452.344: [ImuCal] [NanoSensorCal:GYRO_RPS] Temp Sensitivity: 0.000078, -0.000006, -0.000012
2019-10-22 11:07:30.482 790-972/? I/CHRE: @ 88452.344: [ImuCal] [NanoSensorCal:GYRO_RPS] Temp Intercept: -0.003306, 0.000841, 0.000667
2019-10-22 11:07:33.642 790-972/? I/CHRE: @ 88455.508: [AR_CHRE] still: 100
2019-10-22 11:07:33.982 2298-21383/? I/Checkin: [EventLogChimeraService] Opted in for usage reporting: true
2019-10-22 11:07:33.982 2298-21383/? I/Checkin: [EventLogChimeraService] Aggregate from 1571730045868 (log), 1571730045868 (data)
2019-10-22 11:07:34.012 2033-23324/? E/WakeLock: GCM_CONN_ALARM release without a matched acquire!
2019-10-22 11:07:34.024 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:07:34.024 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:07:34.029 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:07:34.029 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:07:34.032 2298-21383/? I/Checkin: [ServiceDumpSys] dumping service [account]
2019-10-22 11:07:34.037 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:07:34.037 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:07:36.838 790-972/? I/CHRE: @ 88458.695: [AR_CHRE] still: 100
2019-10-22 11:07:36.838 790-972/? I/CHRE: @ 88458.695: [AR_CHRE] send to host, reason=4, sampleSize=112, transitionSize=0, lastAccelTime = 88456121, lastAudioTime = 0 
2019-10-22 11:07:36.839 790-972/? I/CHRE: @ 88458.703: [NanoMinion] sent MSG Type 0x72, Version 0, 1704 bytes, MessageId 153
2019-10-22 11:07:36.841 1162-1989/? V/ContextHubClientManager: Received NanoAppMessage[type = 1025, length = 1712 bytes, broadcast, nanoapp = 0x476f6f676c00100b](data = 0x910006AC 00007200 08C597F6 291A0C08 ...)
2019-10-22 11:07:36.842 1162-1989/? D/ContextHubService: Sending message 1025 version 0 from hubHandle 0, appInstance 0, callBackCount 2
2019-10-22 11:07:40.035 790-972/? I/CHRE: @ 88461.898: [AR_CHRE] still: 100
2019-10-22 11:07:43.238 790-972/? I/CHRE: @ 88465.094: [AR_CHRE] still: 100
2019-10-22 11:07:46.434 790-972/? I/CHRE: @ 88468.297: [AR_CHRE] still: 100

**Comment by abodea** [comment]
I'm not really sure but I believe it's related to this:
```Kotlin
2019-10-22 11:08:45.723 22715-22755/? I/FirefoxAccount: Executing: disconnect
2019-10-22 11:08:45.725 22715-22755/? W/WrappingPersistenceCallback: InternalFxAcct tried persist state, but persistence callback is not set`

**Comment by isabelrios** [comment]
In case it helps, looks like this issue was introduced with commit: https://github.com/mozilla-lockwise/lockwise-android/commit/a36ba93460a8016e89c4352441f07f3518269af7

**Comment by jhugman** [comment]
I can reproduce this with an FxA account, but not the on-device test account.

**Comment by isabelrios** [comment]
same here, on master (or after that commit) with test account the issue is not reproducible...

**Comment by abodea** [comment]
Verified as fixed on **`v3.0.1 (Build 5124)`** using Google Pixel 3a XL(A 9), Samsung Galaxy S10+(A 9).
Please note that I did at least 15 disconnects + re-login and everything worked as expected.
Sometimes when tapping on the disconnect button, it take few seconds(~2-5) until the Welcome Screen with the `Get Started` button is displayed.
@eliserichards @jhugman looks good 🥳!

**Comment by abodea** [comment]
Verified as fixed on **`v3.0.1 (Build 5124)`** using Google Pixel 3a XL(A 9), Samsung Galaxy S10+(A 9).
Please note that I did at least 15 disconnects + re-login and everything worked as expected.
Sometimes when tapping on the disconnect button, it take few seconds(~2-5) until the Welcome Screen with the `Get Started` button is displayed.
@eliserichards @jhugman looks good 🥳!
